mlflow/server/js/src/components/MetricsPlotPanel.js

```diff
@@ -73,12 +77,21 @@ export class MetricsPlotPanel extends React.Component {
     return runs ? JSON.parse(runs).length > 1 : false;
   }
 
-  updateUrlWithSelectedMetrics(selectedMetricKeys) {
+  // Update page URL from component state. Intended to be called after React applies component
+  // state updates, e.g. in a setState callback
+  updateUrlState = (updatedState) => {
```

Comment:
Calls to `setState` replaced with calls to `updateUrlState` (which updates the page URL)